base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -102,9 +122,14 @@ where T: BlockchainBackend
     }
 
     /// Handle inbound blocks from remote nodes and local services.
-    pub async fn handle_block(&self, block: &Block) -> Result<(), CommsInterfaceError> {
-        // TODO: Validate block and create event on block stream
-
-        Ok(())
+    pub async fn handle_block(&mut self, block: &Block) -> Result<(), CommsInterfaceError> {
+        let block_event = match self.blockchain_db.add_block(block.clone()) {
```

Comment:
We need to do additional validation here  before calling add_block, but otherwise this is great. I'll merge it as is for now, but we should assign a ticket to do the full blockchain & contextual validation ASAP.

So now all the `Listening` state in the blockchain FSM needs to do now is listen for `ChainMetaData` messages and respond accordingly.